### DISCUSSION TITLE
Model 'Literal' type with multiple elements showing  up in Swagger page

### DISCUSSION THREAD

**Discussion by ashears** [fastapi/fastapi#8062] [Questions]
**Description**

How can I get a model element with the 'Literal' type to show in the Swagger UI?

As in, 

```
class myModel(BaseModel):
    type: Literal['test']
    test: Literal['test2', 'test3', 'test4']
    count: int
```

This Model shows upon in the swagger page as:
```
        {
          "type": "string",
          "count": 0
        },
```
As in, the 'test' field does not get put into the swagger page

**Comment by ashears** [comment]
Follow up question:
Would it be possible for the Literal type to actually show the allowed element(s) in the SwaggerUI?

**Comment by tiangolo** [comment]
I assume you solved your problem, so thanks for coming back and closing the issue :+1:

**Comment by YuriiMotov** [comment]
Works fine in current version (FastAPI 0.116.0, Pydantic 2.11.7)
![image](https://github.com/user-attachments/assets/c481c7cb-61b5-46d0-8a0f-ba768720829c)
